### DISCUSSION TITLE
What counts as a completion in Copilot Free and how to check usage

### DISCUSSION THREAD

**Discussion by z1645444** [community/community#148154] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Currently I'm using free plan. Copilot is great but there doesn’t seem to be a way to check how many uses I’ve had so far—like a dashboard or tracker.

I'm wondering if triggering a suggestion is considered a completion, or if generating an actual suggestion (regardless of whether it's accepted or not) is what counts. Additionally, does it differentiate between completing a large code block versus a small code snippet?

**Comment by Akash1134** [comment, marked as answer]
Hi @z1645444 ,

Currently, there is no usage tracker available for the Copilot Free plan users. A "completion" is counted when Copilot generates a suggestion in response to your code context or input (e.g., typing code, comments, or invoking it explicitly). It logs completions based on suggestions generated, not what you do with them. Regarding the size of completion, each suggestion generated by Copilot counts as one completion, regardless of size.

[More details about Copilot Free
](https://docs.github.com/en/copilot/managing-copilot/managing-copilot-as-an-individual-subscriber/about-github-copilot-free)

Hope that helps.

**Reply by z1645444** [reply]
Thanks for answering Akash1134 :)

Just curious, do you have plans to develop usage tracker?

**Reply by kayericwinkler** [reply]
So nearly every char I type is one completion (as the suggestions change while I type) -- no wonder that the free credits are gone in no time. The "free" option is more like a brief trial, then.

**Reply by D00pN00dle** [reply]
A suggestion is a completion??

Wow. That's wild stuff. 

Bad. bad. bad.

**Reply by adam3141** [reply]
Yeah - that's really bad. The VSCode page says it's about 80 per day and they call that plenty. If I am typing code, I can see the suggestions pop up and I would run through 80 in about 30 minutes give or take.

I think 2000 accepted completions a month would be a lot better

**Reply by lfavole** [reply]
Good idea, but it takes AI compute power to generate completions, whether they are accepted or not.
To count accepted completions, the program should report to the server that a completion was accepted, so this could be tricked and abused to (maybe) have unlimited completions.

They should really add a counter: for example, you can type `Alt+Space`, it connects to GitHub Copilot and sends back the completion.
This would also remove the useless completions that pop up when typing.

**Reply by senese** [reply]
They should just be crystal clear about what is going on. Changing the term from `Code Completions` to `Code Suggestions` is enough.

**Comment by TsengSR** [comment]
Pretty lame indeed. It should be selected suggestions, as code that's useless or wrong or just happens before you finished typing is useless for the user and shouldn't get conunted towards the quota

**Comment by rnickerson-levy** [comment]
Yeah, this kind of quota system is effectively useless as I'd blow through 2k "suggestions" in an afternoon easily. Uninstalled the extension entirely (since there seems to be no way to just turn it off).

It'd be different if I could ask for a completion suggestion. I was setting up a new NextJS project and every time I created a new page it would suggest the basically empty boilerplate, which is nice enough, but I can achieve better results with snippets and I didn't see any way to prevent wasting my credits on useless code generation. All of which make me entirely dubious of ever actually paying for the service, especially when those plans also include paying for uncontrollable and unwanted suggestions.

**Reply by luisAfonsoM** [reply]
VSCode does not support this I believe, but Intellij has enable and disable feature

**Comment by D00pN00dle** [comment]
As far as the free plan goes, it isn't really optimal but I think the pro plan is more than reasonable if you're writing code daily.

**Comment by charlielee** [comment]
While your usage isn't shown in the UI for the free plan, you can see your remaining usage by opening the network dev tools on https://github.com/settings/copilot and looking at the response to `GET https://github.com/github-copilot/chat/entitlement`.

<img width="511" alt="image" src="https://github.com/user-attachments/assets/2bb90313-f3fb-4335-8b1d-b3a3a54afd50" />

**Reply by luisAfonsoM** [reply]
Was this perhaps changed already?
![image](https://github.com/user-attachments/assets/4a7050d3-59f7-4d6e-b666-b09d69e11aaa)

**Reply by z1645444** [reply]
> Was this perhaps changed already? ![image](https://private-user-images.githubusercontent.com/126819903/424754181-4a7050d3-59f7-4d6e-b666-b09d69e11aaa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI1MjMzNTUsIm5iZiI6MTc0MjUyMzA1NSwicGF0aCI6Ii8xMjY4MTk5MDMvNDI0NzU0MTgxLTRhNzA1MGQzLTU5ZjctNGQ2ZS1iNjY2LWIwOWQ2OWUxMWFhYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMyMVQwMjEwNTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lYzFhOTlhNjc4MmRlZDBhM2ExNmE3MDdjOGY4ZGZkNTg0ZDJhNGNkNjI3ZGMwODAwZDdhNDFhNzExYWE0OWUxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.nlSDpPEcPLUd2FD3i3g2SZzAcMaNOpTN2jG39my-2N0)

No, I just checked and it does exist. Try to re-login maybe it will work luisAfonsoM :)

**Reply by lfavole** [reply]
Maybe you should enable this setting...
!["Show Copilot" setting](https://github.com/user-attachments/assets/fcb73871-800e-408a-8085-558e5ee7fba1)

**Reply by harsh183** [reply]
this isn't working for me on "GitHub Copilot Business is active for your account". I tried logging in again and disabling uBlockOrigin (which was blocking a few Github tracking requests). 

Perhaps we'll wait till they actually release this feature :sigh:

**Comment by leelasaivoonna** [comment]
<img width="236" height="388" alt="image" src="https://github.com/user-attachments/assets/3ddd9f2d-2924-4d61-afaf-2c3ab1e96e30" />
@z1645444 
 You can see your usage at bottom of the VSCode, but it would show tracker on monthly basis
